src/sentry/snuba/metrics/mqb_query_transformer.py

```diff
@@ -252,9 +267,10 @@ def _derive_mri_to_apply(project_ids, select, orderby):
     has_order_by_team_key_transaction = False
     if orderby is not None:
         for orderby_field in orderby:
-            if orderby_field.field.op == TEAM_KEY_TRANSACTION_OP:
-                has_order_by_team_key_transaction = True
-                break
+            if isinstance(orderby_field.field, MetricField):
+                if orderby_field.field.op == TEAM_KEY_TRANSACTION_OP:
```

Comment:
if you want you can combine it with `and`:
```
isinstance(orderby_field.field, MetricField) and orderby_field.field.op == TEAM_KEY_TRANSACTION_OP:
```
but may be it can be less readable.